OpenRA.Mods.Common/Traits/Buildings/BaseProvider.cs

```diff
@@ -74,7 +78,7 @@ public bool Ready()
 
 		bool ValidRenderPlayer()
 		{
-			return self.Owner == self.World.RenderPlayer || (allyBuildEnabled && self.Owner.IsAlliedWith(self.World.RenderPlayer));
+			return buildRadiusEnabled ? self.Owner == self.World.RenderPlayer || (allyBuildEnabled && self.Owner.IsAlliedWith(self.World.RenderPlayer)) : false;
```

Comment:
I think this can be simplified to `return buildRadiusEnabled && (self.Owner == self.World.RenderPlayer || (allyBuildEnabled && self.Owner.IsAlliedWith(self.World.RenderPlayer)));`.